add modified dockerfiles to the list of plays

syed-awais-ali opened the pull request:
Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] Have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?

syed-awais-ali commented:
@edx/devops Review please. Thanks

feanil commented:
@awais786 can you show the output of some tests you've run on this?  What's the output when you pass in a set of dockerfiles as the changeset.

syed-awais-ali commented:
@feanil Made another branch for testing, by giving old and new commits to it and setting the value of `TRAVIS_BUILD_DIR` to configuration repo. I am on it, will paste them shortly.

feanil commented:
@syed-awais-ali you can also just run the python script manually passing in a list of files.

syed-awais-ali commented:
I changed certain files to test the difference between two commits. I changed two dockerfiles and two playbooks and pasted the output below.

## Output
```
set(['docker/build/xqueue/Dockerfile', 'playbooks/create_role.yml', 'docker/build/analytics_api/Dockerfile', 'playbooks/edx-east/alton.yml'])
```

feanil commented:
@syed-awais-ali I ran the following tests on master and on your branch and see some issues.

# test_data.txt
```
CHANGELOG.md
playbooks/callback_plugins/task_timing.py
playbooks/roles/edxapp/defaults/main.yml
playbooks/roles/forum/defaults/main.yml
playbooks/roles/go-server/tasks/main.yml
playbooks/roles/jenkins_worker/defaults/main.yml
playbooks/roles/jenkins_worker/tasks/python_platform_worker.yml
playbooks/roles/mongo_client/defaults/main.yml
playbooks/roles/tools_jenkins/defaults/main.yml
docker/plays/discovery.yml
docker/build/ecommerce/Dockerfile
```
On Master:
```
~/src/configuration$ cat test_data.txt | python util/parsefiles.py
edxapp forum
```

On this branch:
```
~/src/configuration$ cat test_data.txt | python util/parsefiles.py
set(['playbooks/callback_plugins/task_timing.py', 'playbooks/roles/tools_jenkins/defaults/main.yml', 'docker/build/ecommerce/Dockerfile', 'docker/plays/discovery.yml', 'CHANGELOG.md', 'playbooks/roles/edxapp/defaults/main.yml', 'playbooks/roles/jenkins_worker/defaults/main.yml', 'playbooks/roles/jenkins_worker/tasks/python_platform_worker.yml', 'playbooks/roles/forum/defaults/main.yml', 'playbooks/roles/go-server/tasks/main.yml', 'playbooks/roles/mongo_client/defaults/main.yml'])
edxapp forum
```

Expected results after the correct change:
```
~/src/configuration$ cat test_data.txt | python util/parsefiles.py
edxapp forum discovery ecommerce
```

It should not print the `repr` of the set.  It should out put the names of the roles if they are in the parsefiles_config.yml file.

syed-awais-ali commented:
Tested with the following input
```
playbooks/callback_plugins/task_timing.py
docker/build/xqueue/Dockerfile
playbooks/roles/edxapp/defaults/main.yml
playbooks/roles/forum/defaults/main.yml
playbooks/roles/go-server/tasks/main.yml
playbooks/roles/jenkins_worker/defaults/main.yml
docker/build/xqwatcher/Dockerfile
playbooks/roles/jenkins_worker/tasks/python_platform_worker.yml
playbooks/roles/mongo_client/defaults/main.yml
playbooks/roles/tools_jenkins/defaults/main.yml
docker/plays/discovery.yml
docker/build/ecommerce/Dockerfile
docker/build/analytics_api/Dockerfile
```

Output
```
edxapp forum
analytics_api xqueue ecommerce xqwatcher

```

feanil commented:
@syed-awais-ali all the roles should be space separated output in one line.

syed-awais-ali commented:
@feanil done. Review please

syed-awais-ali commented:
## Ouput
```
edxapp forum analytics_api xqueue ecommerce xqwatcher
```

feanil commented:
@syed-awais-ali it looks like your code only cares if the Dockerfiles are in the list.  We actually care about any file under 'docker/build/*/` folders.  Because there can be other files next to the docker files or subdirectories under a particular apps folder that have changes which would indicate that the container needs to be rebuilt.

syed-awais-ali commented:
## Input file
```
playbooks/callback_plugins/task_timing.py
docker/build/xqueue/Dockerfile
docker/build/analytics_api
playbooks/roles/edxapp/defaults/main.yml
playbooks/roles/forum/defaults/main.yml
playbooks/roles/go-server/tasks/main.yml
playbooks/roles/jenkins_worker/defaults/main.yml
docker/build/xqwatcher/Dockerfile
playbooks/roles/jenkins_worker/tasks/python_platform_worker.yml
playbooks/roles/mongo_client/defaults/main.yml
playbooks/roles/tools_jenkins/defaults/main.yml
docker/plays/discovery.yml
docker/build/ecommerce/Dockerfile
docker/build/analytics_api/Dockerfile
docker/build/chrome/Dockerfile
docker/build/credentials/ansible_overrides.yml
docker/build/analytics_api/ansible_overrides.yml
docker/build/ecomworker/Dockerfile
docker/build/ecomworker/ansible_overrides.yml
docker/build/rabbitmq/sample.json
```

## Output
```
edxapp forum ecomworker chrome rabbitmq analytics_api xqwatcher credentials xqueue ecommerce
```

feanil commented on util/parsefiles.py:
```diff
@@ -330,6 +330,38 @@ def _get_role_name(role):
+    # get individual parts of a file path
+    dirs = path.parts
+    # name of play
+    return dirs[-2]
```
This only works if all the files in the same directory as the Dockerfile.  There can be sub-directories under here as well.

feanil commented on util/parsefiles.py:
```diff
@@ -330,6 +330,38 @@ def _get_role_name(role):
+    return items
+
+
+def _get_play_name_from_dockerfile(path):
```
This name and the description below are in correct now that we are looking at all of the files under the `docker/build/<x>` directories.
